pages/blog/integrating-mailchimp-with-your-nodejs-app.mdx

```diff
@@ -0,0 +1,350 @@
+
+3. Next up, we need to specify the folder which contains your frontend application and connect it to the express server. Here, we’re having a folder named `public`, which includes all our frontend code.
+
+```
+app.use(express.static(path.join(__dirname, 'public')))
+```
+
+4. Perfect, let's create our `POST` request quickly. We're just setting it up to add a `console.log()` to check if everything's working as expected.
+
+```
+app.post('/subscribe', (req, res) => {
+    console.log(“I am listening”);
+})
+```
+
+5. Now let's define the `PORT` which will host our application.
+
+```
+const PORT = process.env.PORT || 8888;
+```
+
+This sets the value to the environment `PORT` value in case we're using a cloud hosting or default it to 8888, which is our local machine endpoint.
+
+6. Finally, for all this to work, let's start a socket that listens for our connections on the `PORT`.
+
+```
+app.listen(PORT, console.log('The server has started!'));
+```
+
+7. Woof! That's so much setup for just a basic API request. Let's test our setup up till now. We'll be using the node command for this.
+
+```
+node app
+```
+
+As soon as you run it, you will notice that the terminal says
+
+```
+The server has started!
+```
+
+You can also check your frontend to start a post request by entering the email field. As soon as you do that, you'll notice the terminal says
+
+```
+I am listening
+```
+
+This proves that our server is up and running, and the post request has also gone through. If you do not have a frontend and want to test this, you can easily use a tool like Postman and send a `POST` request similar to the one we have mentioned above for the frontend application.
+
+Now finally, it is time to do some real fun stuff! Let’s integrate this with MailChimp!
+
+## Connecting your Node.js App to MailChimp
+
+For this whole process, we’re directly referring to the [MailChimp API Documentation](https://mailchimp.com/developer/marketing/api/list-activity/). They have excellent documentation stating all the different things you need to be aware of with extremely detailed resources. Make sure to check that out if you need any additional information that is not mentioned here directly.
+
+We’re using the standard URL route here. You can also download the [MailChimp NodeJS module](https://www.npmjs.com/package/mailchimp-api-v3) if you want to as well. Although this makes your life easier in implementing the API, it adds many other modules that we do not need for this particular project.
+
+### Getting your API Endpoint
+
+So the first thing here is to get the URL of your API. If you check the documentation, the URL looks something like `https://<dc>.api.mailchimp.com/3.0`
+
+Here the `<dc>` corresponds to the data center applicable for your account. If you go back to _step 4_ of **Getting your API Key**, you can see that for us, the data center is `us6`. Hence for us, the URL looks something like `https://us6.api.mailchimp.com/3.0`
+
```

Comment:
This feels ... not right. You're asking the user to go back to earlier in the post and grab another value. At the very least this should read linearly so that you're either telling them what to do here or made note of the value when the user was following step 4.

But even making those adjustments feels a little weird. If you use node-mailchimp, you don't need to know this value. You can rely on the library to manage that for you.

Reply:
I feel using the library is not a smart choice for such a basic integration blog. If we were diving deeper into accessing different aspects of Mailchimp using their module, it might have made a lot of sense.

For this step, I have clearly mentioned in the section of getting your API Key that you need to be mindful of this value. I've also repeated myself here so that anyone who's reading this might not have to go back and forth.